packages/@aws-cdk/aws-sagemaker/lib/endpoint.ts

```diff
@@ -0,0 +1,448 @@
+import { EOL } from 'os';
+import * as appscaling from '@aws-cdk/aws-applicationautoscaling';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { EndpointConfig, IEndpointConfig, InstanceProductionVariant } from './endpoint-config';
+import { InstanceType } from './instance-type';
+import { sameEnv } from './private/util';
+import { CfnEndpoint } from './sagemaker.generated';
+import { ScalableInstanceCount } from './scalable-instance-count';
+
+/*
+ * Amazon SageMaker automatic scaling doesn't support automatic scaling for burstable instances such
+ * as T2, because they already allow for increased capacity under increased workloads.
+ * https://docs.aws.amazon.com/sagemaker/latest/dg/endpoint-auto-scaling-add-console.html
+ */
+const BURSTABLE_INSTANCE_TYPE_PREFIXES = Object.entries(ec2.InstanceClass)
+  .filter(([name, _]) => name.startsWith('T'))
+  .map(([_, prefix]) => `ml.${prefix}.`);
```

Comment:
So I see `T3`, `T3a`, `T2`, and `T4G` are the burstable instances that do not support automatic scaling from your link. And here we are assuming that all `T` prefixed instances are burstable. I know nothing about ec2 instances so I'm going to take your word for it, but it would be a problem if AWS released `T5` tomorrow that wasn't burstable... I'm assuming thats unlikely

Reply:
The following snippet of [EC2 documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/burstable-performance-instances.html) implies that all `T` instances will be burstable:

> These low-to-moderate CPU utilization workloads lead to wastage of CPU cycles and, as a result, you pay for more than you use. To overcome this, you can leverage the low-cost burstable general purpose instances, which are the T instances.
> 
> The T instance family provides a baseline CPU performance with the ability to burst above the baseline at any time for as long as required.